radixdlt/src/main/java/com/radixdlt/consensus/EventCoordinator.java

```diff
@@ -1,222 +1,28 @@
-		if (!proposerElection.isValidProposer(selfKey.getUID(), vote.getVertexMetadata().getView())) {
-			log.warn(String.format("%s Ignoring confused vote %s for %s", getShortName(), vote.hashCode(), vote.getVertexMetadata().getView()));
-			return;
-		}
-
-		// accumulate votes into QCs in store
-		Optional<QuorumCertificate> potentialQc = this.pendingVotes.insertVote(vote);
-		if (potentialQc.isPresent()) {
-			QuorumCertificate qc = potentialQc.get();
-			log.info(this.getShortName() + ": Formed QC: " + qc);
-			this.processQC(qc);
-		}
-	}
-
-	public void processNewView(NewView newView) {
-		log.info(this.getShortName() + ": Processing NEW_VIEW_MESSAGE: " + newView);
-
-		// only do something if we're actually the leader for the next view
-		if (!proposerElection.isValidProposer(selfKey.getPublicKey().getUID(), newView.getView())) {
-			log.warn(String.format("Got confused new-view %s for view ", newView.hashCode()) + newView.getView());
-			return;
-		}
-
-		this.processQC(newView.getQC());
-
-		this.pacemaker.processNewView(newView)
-			.ifPresent(this::startQuorumNewView);
-	}
-
-	public void processProposal(Vertex proposedVertex) {
-		log.info(this.getShortName() + ": Processing PROPOSAL_MESSAGE: " + proposedVertex);
-
-		final View currentView = this.pacemaker.getCurrentView();
-		if (proposedVertex.getView().compareTo(currentView) < 0) {
-			log.info("Ignore proposal current " + currentView + " but proposed " + proposedVertex.getView());
-			return;
-		}
-
-		processQC(proposedVertex.getQC());
-
-		// TODO: Sync at this point
-
-		final View updatedView = this.pacemaker.getCurrentView();
-		if (proposedVertex.getView().compareTo(updatedView) != 0) {
-			log.info("Ignore proposal current " + updatedView + " but proposed " + proposedVertex.getView());
-			return;
-		}
-
-		try {
-			vertexStore.insertVertex(proposedVertex);
-		} catch (VertexInsertionException e) {
-			log.info("Rejected vertex insertion " + e);
-
-			// TODO: Better logic for removal on exception
-			final Atom atom = proposedVertex.getAtom();
-			if (atom != null) {
-				mempool.removeRejectedAtom(atom.getAID());
-			}
-			return;
-		}
-
-		try {
-			final Vote vote = safetyRules.voteFor(proposedVertex);
-			final EUID leader = this.proposerElection.getProposer(updatedView);
-			log.info(this.getShortName() + ": Sending Vote to " + this.getShortName(leader) + ": " + vote);
-			networkSender.sendVote(vote, leader);
-		} catch (SafetyViolationException e) {
-			log.error("Rejected " + proposedVertex, e);
-		}
-
-		// TODO: Proceed to next view if not leader or next leader
-		// TODO: For now, just depend on Timeout events
-	}
-
-	public void processLocalTimeout(View view) {
-		log.info(this.getShortName() + ": Processing LOCAL_TIMEOUT: " + view);
-
-		// proceed to next view if pacemaker feels like it
-		this.pacemaker.processLocalTimeout(view)
-			.ifPresent(this::proceedToView);
-	}
+public interface EventCoordinator {
+	void processVote(Vote vote);
```

Comment:
Really should have comments for these methods.  Added bonus is that they will be inherited in implementing classes.